### PR TITLE
fix: update to latest Cloudsmith API bindings

### PR DESCRIPTION
**Pull request by paddycarey**
Updates the provider to use the latest version of the Cloudsmith API bindings.

### CODE REVIEW

File: cloudsmith/data_source_package_list.go [hunk 1]
```diff
@@ -96,30 +100,27 @@ func dataSourcePackageListRead(d *schema.ResourceData, m interface{}) error {
...
+		pkg["name"] = packageItem.GetName()
+		pkg["slug"] = packageItem.GetSlug()
+		pkg["slug_perm"] = packageItem.GetSlugPerm()
+		pkg["format"] = packageItem.GetFormat()
+		pkg["version"] = packageItem.GetVersion()
+		pkg["is_sync_awaiting"] = packageItem.GetIsSyncAwaiting()
+		pkg["is_sync_completed"] = packageItem.GetIsSyncCompleted()
+		pkg["is_sync_failed"] = packageItem.GetIsSyncFailed()
+		pkg["is_sync_in_progress"] = packageItem.GetIsSyncInProgress()
+		pkg["is_sync_in_flight"] = packageItem.GetIsSyncInFlight()
```
Comment (lskillen) [inline cloudsmith/data_source_package_list.go:119]: NAB: Should we put a status in for `is_quarantined` (ofc that's an enhancement 😆)

File: cloudsmith/resource_repository.go [hunk 2]
```diff
@@ -104,73 +92,66 @@ func resourceRepositoryRead(d *schema.ResourceData, m interface{}) error {
...
 		return err
 	}
 
-	d.SetId(repository.SlugPerm)
+	d.SetId(repository.GetSlugPerm())
+
+	checkerFunc := func() error {
+		// this is somewhat of a hack until we have a better way to poll for a
+		// repository being updated (changes incoming on the API side)
+		time.Sleep(time.Second * 5)
```
Comment (lskillen) [inline cloudsmith/resource_repository.go:115]: Unholy Necessity

**Review by lskillen — APPROVED** [review]
Looks pretty good to me :) w/ 1 question + NAB + Unholy GIF, each.

File: cloudsmith/data_source_package_list.go (hunk 1 above)
Comment (paddycarey) [reply cloudsmith/data_source_package_list.go:119]: Yep, will do in a follow up!

**Pull request merged**